core/src/apps/management/recovery_device/recover.py

```diff
@@ -48,6 +54,12 @@ def _process_slip39(words: str) -> Optional[bytes]:
         storage.recovery.set_slip39_threshold(threshold)
         storage.recovery.set_remaining(threshold - 1)
         storage.recovery_shares.set(index, words)
+        if group_count > 1:
+            storage.recovery.set_slip39_group_count(group_count)
+            storage.recovery.set_slip39_group_threshold(group_threshold)
+            shares_remaining = bytearray([0x10] * group_count)
+            shares_remaining[group_index] = threshold - 1
```

Comment:
We can provide functions in storage for this.

Reply:
fixed with the storage refactor